src/ctap/storage.rs

```diff
@@ -380,44 +395,117 @@ impl PersistentStore {
+            None => MAX_PIN_RETRIES,
+            Some((_, entry)) => {
+                debug_assert_eq!(entry.data.len(), 1);
+                entry.data[0]
+            }
+        }
     }
 
     pub fn decr_pin_retries(&mut self) {
-        let (index, old_value) = self.pin_retries_entry();
-        let new_value = old_value.saturating_sub(1);
-        self.store
-            .replace(
-                index,
-                StoreEntry {
-                    tag: PIN_RETRIES,
-                    data: &[new_value],
-                    sensitive: false,
-                },
-            )
-            .unwrap();
+        match self.store.find_one(&Key::PinRetries) {
+            None => {
+                self.store
+                    .insert(StoreEntry {
+                        tag: PIN_RETRIES,
+                        data: &[MAX_PIN_RETRIES.saturating_sub(1)],
+                        sensitive: false,
+                    })
+                    .unwrap();
+            }
+            Some((index, entry)) => {
+                debug_assert_eq!(entry.data.len(), 1);
+                if entry.data[0] == 0 {
+                    return;
+                }
+                let new_value = entry.data[0].saturating_sub(1);
+                self.store
+                    .replace(
+                        index,
+                        StoreEntry {
+                            tag: PIN_RETRIES,
+                            data: &[new_value],
+                            sensitive: false,
+                        },
+                    )
+                    .unwrap();
+            }
+        }
     }
 
     pub fn reset_pin_retries(&mut self) {
-        let (index, _) = self.pin_retries_entry();
+        if let Some((index, _)) = self.store.find_one(&Key::PinRetries) {
+            self.store.delete(index).unwrap();
+        }
+    }
+
+    #[cfg(feature = "with_ctap2_1")]
+    pub fn min_pin_length(&self) -> u8 {
         self.store
-            .replace(
-                index,
-                StoreEntry {
-                    tag: PIN_RETRIES,
-                    data: &[MAX_PIN_RETRIES],
-                    sensitive: false,
-                },
-            )
-            .unwrap();
+            .find_one(&Key::MinPinLength)
+            .map_or(DEFAULT_MIN_PIN_LENGTH, |(_, entry)| entry.data[0])
+    }
+
+    #[cfg(feature = "with_ctap2_1")]
+    pub fn set_min_pin_length(&mut self, min_pin_length: u8) {
+        let entry = StoreEntry {
+            tag: MIN_PIN_LENGTH,
+            data: &[min_pin_length],
+            sensitive: false,
+        };
+        match self.store.find_one(&Key::MinPinLength) {
+            None => {
+                self.store.insert(entry).unwrap();
+            }
+            Some((index, _)) => {
+                self.store.replace(index, entry).unwrap();
+            }
+        }
+    }
+
+    #[cfg(feature = "with_ctap2_1")]
+    pub fn _min_pin_length_rp_ids(&self) -> Vec<String> {
```

Comment:
Same here, is this unused, or used for tests only?
Same for the next function.

Reply:
As mentioned above, it's a future feature that belongs to this change when you read the specification.